### PR TITLE
Fix Xcode 16 build error

### LINKED ISSUE #1209: `Ambiguous inference of Objective-C name for instance method` when upgrading to FirebaseAuthUI 14.2.4
### Step 2: Describe your environment

  * Objective C or Swift:  Swift 6
  * iOS version: iOS 18.0.1
  * Firebase SDK version: 11.3.0
  * FirebaseUI version: 14.2.4
  * CocoaPods Version: 1.15.2
  
### Step 3: Describe the problem:

Screenshot 2024-10-09 at 11 14 27 PM

```
    func authUI(_ authUI: FUIAuth, didSignInWith authDataResult: AuthDataResult?, error: Error?) {
```

After updating to XCode 16, Swift 6, Firebase 11.3.0, FirebaseUI 14.2.4, I can't get past this error. 
This code worked fine in XCode 15, Swift 5.5, Firebase 10.29.0, FirebaseUI 14.0.0.

What's the correct `func` invocation for that method?

Imports above are:

```
import UIKit
import FirebaseAuth
import FirebaseEmailAuthUI
import FirebaseOAuthUI
import FirebaseGoogleAuthUI
import FirebaseAnalytics
```

### PR DESCRIPTION
**Pull request by paulb777**
Fix #1209 

Recent versions of Xcode 16 give compilation errors for the FUIAuthDelegate Firebase 10 compatibility methods. This PR resolves the build issue by deleting those methods. 

Firebase 10 users of  FUIAuthDelegate should use a a FirebaseUI version earlier than 14.2.5.

### CODE REVIEW

File: FirebaseAuthUI.podspec [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |s|
   s.name         = 'FirebaseAuthUI'
-  s.version      = '14.2.4'
```
Comment (morganchen12) [inline FirebaseAuthUI.podspec:3]: Can you update the FirebaseAuth dependency in this file too? (Line 21)

**Review by morganchen12 — APPROVED** [review]
LGTM with one requested change.

File: FirebaseAuthUI.podspec (hunk 1 above)
Comment (paulb777) [reply FirebaseAuthUI.podspec:3]: Good catch!

**Comment by paulb777** [comment]
Removed Podfile.lock's so CI tests with latest acceptable dependencies

**Pull request merged**